### PR TITLE
update to support page down and get issue count with button A

### PR DESCRIPTION
**Pull request by erich-wang**

### CODE REVIEW

File: config.h [hunk 1]
```diff
@@ -1,7 +1,4 @@
-#define INTERVAL 3000
-
 #define RGB_LED_BRIGHTNESS 32
 
-#define DEVICE_ID "AZ3166"
-
-#define CONNECTION_STRING ""
\ No newline at end of file
+//Need to specify device connection string here: "HostName=**;DeviceId=**;SharedAccessKey=**"
+#define CONNECTION_STRING ""
```
Comment (formulahendry) [inline config.h]: Is this project only for demo purpose? If the project will officially go to the https://microsoft.github.io/azure-iot-developer-kit/docs/projects/, then we need to read/write the CONNECTION_STRING into EEPROM.

File: utility.cpp [hunk 2]
```diff
@@ -30,6 +78,41 @@ void blinkLED()
...
+{
+    Screen.clean();
+    Screen.print("Requesting...");
+    Serial.println("Requesting...");
+
+    bool success = false;
+
+    HTTPClient *client = new HTTPClient(GITHUB_CERT, HTTP_GET, REPO_API_URL);
+    client->set_header("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64)");
+    const Http_Response *response = client->send();
```
Comment (SLdragon) [inline utility.cpp]: Need delete or free ?

File: utility.cpp [hunk 3]
```diff
@@ -42,25 +125,24 @@ static IOTHUBMESSAGE_DISPOSITION_RESULT c2dMessageCallback(IOTHUB_MESSAGE_HANDLE
...
         blinkLED();
-        free(temp);
         return IOTHUBMESSAGE_ACCEPTED;
     }
 }
 
 void iothubInit()
 {
     srand((unsigned int)time(NULL));
+    msgStart = msgBodyLen = 0;
```
Comment (michaeljqzq) [inline utility.cpp]: line #143, if rand() is not used in the code, srand can be removed

File: utility.cpp [hunk 4]
```diff
@@ -89,5 +171,13 @@ void iothubInit()
 
 void iothubLoop(void)
```
Comment (michaeljqzq) [inline utility.cpp]: Better keep inform with previous function without parameter. iothubLoop()

File: utility.cpp [hunk 5]
```diff
@@ -30,6 +78,41 @@ void blinkLED()
...
+void getRepoIssues()
+{
+    Screen.clean();
+    Screen.print("Requesting...");
+    Serial.println("Requesting...");
+
+    bool success = false;
+
+    HTTPClient *client = new HTTPClient(GITHUB_CERT, HTTP_GET, REPO_API_URL);
+    client->set_header("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64)");
```
Comment (formulahendry) [inline utility.cpp]: We'd better not pretend as a browser.
Per https://developer.github.com/v3/#user-agent-required, we'd better use  the name of our application.

File: utility.cpp [hunk 6]
```diff
@@ -30,6 +78,41 @@ void blinkLED()
...
+            msgBodyLen = strlen(msgBody);
+            msgStart = 0;
+            displayIssue();
+            success = true;
+        }
+    }
+    if(!success)
+    {
+        Screen.clean();
+        Screen.print("Failed to get issues.");
```
Comment (SummerSun) [inline utility.cpp]: make sure it could show in the screen

File: utility.cpp [hunk 7]
```diff
@@ -42,25 +125,24 @@ static IOTHUBMESSAGE_DISPOSITION_RESULT c2dMessageCallback(IOTHUB_MESSAGE_HANDLE
...
-        memcpy(temp, buffer, size);
-        temp[size] = '\0';
-        LogInfo("Receive C2D message: %s", temp);
-        showC2DMessageReceived(temp);
+        //TODO: handle if message length >= 2048
+        msgStart = 0;
+        msgBody[size] = '\0';
+        memcpy(msgBody, buffer, size);
+        msgBodyLen = size;
+        LogInfo("Receive C2D message: %s", msgBody);
```
Comment (SummerSun) [inline utility.cpp:129]: redundancy, loginfo is implemented with serial.println

File: utility.cpp [hunk 8]
```diff
@@ -7,15 +7,63 @@
...
 #include "config.h"
 #include "utility.h"
+#include "json.h"
+#include "http_client.h"
 
 static IOTHUB_CLIENT_LL_HANDLE iotHubClientHandle;
-static HTS221Sensor *ht_sensor;
 static RGB_LED rgbLed;
+static int msgStart;
+static char msgBody[2048];
```
Comment (SummerSun) [inline utility.cpp:16]: do we need this large

File: utility.cpp (hunk 2 above)
Comment (SummerSun) [inline utility.cpp]: suggest not const inside function

File: utility.cpp [hunk 9]
```diff
@@ -30,6 +78,41 @@ void blinkLED()
...
+    if(response != NULL)
+    {
+        json_object *jsonObject, *countObject;
+        int issueCount;
+        Serial.println(response->body);
+        if((jsonObject = json_tokener_parse(response->body)) != NULL &&
+            (countObject = json_object_object_get(jsonObject, "open_issues_count")) != NULL &&
+            (issueCount = json_object_get_int(countObject)) >= 0)
+        {
+            sprintf(msgBody, "There're total %d open issues in repo %s", issueCount, REPO_URL);
```
Comment (formulahendry) [inline utility.cpp]: issue(s)

File: utility.cpp (hunk 2 above)
Comment (erich-wang) [reply utility.cpp]: no need, it will be deleted when deleting client

File: utility.cpp (hunk 3 above)
Comment (erich-wang) [reply utility.cpp]: good catch

File: utility.cpp (hunk 6 above)
Comment (erich-wang) [reply utility.cpp]: good catch

**Review by michaeljqzq — APPROVED** [review]

**Pull request merged**